system_tests/datastore.py

```diff
@@ -271,7 +287,13 @@ def test_projection_query(self):
         self.assertEqual(catelyn_stark_dict,
                          {'name': 'Catelyn', 'family': 'Stark'})
 
-        catelyn_tully_entity = entities[3]
+        if EMULATOR_DATASET is None:
```

Comment:
I think using your custom indexes is good because it makes your emulator usage match production usage.

But I think it's also not entirely necessary as long as you use queries which cannot choose which index to use.